### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by mraszyk**
This MR makes the following updates to the README:
- update links to Interface Spec
- drop ghcid instructions (ghcid might not work any longer)
- drop cabal repl instructions (I could not reproduce them anymore)
- using ic-ref with dfx
- replace 0.0.0.0 by localhost
- fix typos

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,7 +149,16 @@ functional tests against it.
 
 ### Usage
 
-Pass `--endpoint http://localhost:8080/` to run against a specific node.
+Before running `ic-ref-test`, make sure you have built the universal canister
+(using `nix-shell`):
+```
+  cd universal-canister
+  nix-shell --command 'cargo build --target wasm32-unknown-unknown --release'
+```
```
Comment (dsd-at-dfinity) [inline README.md:157]:
```suggestion
```console
$ cd universal-canister
$ nix-shell --command 'cargo build --target wasm32-unknown-unknown --release'
```
```

File: README.md [hunk 2]
```diff
@@ -161,41 +187,14 @@ calculates its request id.
...
-
+The modules of the above tools can be used for other purposes. In that sense, the whole project is one big Haskell library that can be used in quick experiments or as a libary to build other tools (e.g. a test framework for developer tools as [in the case of DFX](https://github.com/dfinity/sdk/blob/master/.github/workflows/e2e.yml)).
 
 Continuous Integration
 ----------------------
 
 We use GitHub Actions to trigger builds of the jobs defined in `./default.nix`. However the builds themselves are run on the [nixbuild.net](https://nixbuild.net/) service since it provides more capacity and is more efficient than GitHub runners.
 
-Please use the artifacts produced by GitHub Actions and nixbuild.net at your own risk or consider building independently from source.
+Please use the artifacts produced by GitHub Actions and [nixbuild.net](https://nixbuild.net/) at your own risk or consider building independently from source.
```
Comment (dsd-at-dfinity) [inline README.md:197]:
```suggestion
Please use the artifacts produced by GitHub Actions and [nixbuild.net](https://nixbuild.net/) at your own risk or consider building them independently from source.
```

**Review by dsd-at-dfinity — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -223,21 +222,6 @@ Running `nix-shell` gives you an environment that allows you to build the
...
-
-The symbolic link in `test-data/universal-canister.wasm` points to the
-build output produced by
-
-    cd universal-canister
-    nix-shell --command 'cargo build --target wasm32-unknown-unknown --release'
-
 You can now run the test suite from the top-level directory with
 
     cabal run ic-ref-test
```
Comment (nomeata) [inline README.md:227]: Is building the universal canister no longer needed to run `ic-ref-test`?

Comment (mraszyk) [reply README.md:227]: I moved the instructions higher up to the beginning of Usage section to make them more visible.

**Pull request merged**